addons/mail/static/src/core/common/thread_model.js

```diff
@@ -354,7 +354,11 @@ export class Thread extends Record {
     }
 
     get displayToSelf() {
-        return this.is_pinned || (["channel", "group"].includes(this.type) && this.hasSelfAsMember);
+        return (
+            this.is_pinned ||
+            (this.isChatChannel && this.hasSelfAsMember) ||
+            this.eq(this._store.discuss.thread)
```

Comment:
comparison to discuss should be moved outside, as displayToSelf can be used in other context where this makes no sense, such as messaging menu